src/OpenTelemetry.Exporter.InMemory/InMemoryExporterMetricsExtensions.cs

```diff
@@ -93,5 +135,40 @@ public static MeterProviderBuilder AddInMemoryExporter(this MeterProviderBuilder
 
             return builder.AddReader(metricReader);
         }
+
+        private static MeterProviderBuilder AddInMemoryExporter(
+            MeterProviderBuilder builder,
+            ICollection<MetricSnapshot> exportedItems,
+            MetricReaderOptions metricReaderOptions,
+            Action<MetricReaderOptions> configureMetricReader)
+        {
+            configureMetricReader?.Invoke(metricReaderOptions);
+
+            var metricExporter = new InMemoryExporter<Metric>(
+                exportFunc: metricBatch => ExportMetricSnapshot(metricBatch, exportedItems));
+
+            var metricReader = PeriodicExportingMetricReaderHelper.CreatePeriodicExportingMetricReader(
+                metricExporter,
+                metricReaderOptions,
+                DefaultExportIntervalMilliseconds,
+                DefaultExportTimeoutMilliseconds);
+
+            return builder.AddReader(metricReader);
+        }
+
+        private static ExportResult ExportMetricSnapshot(in Batch<Metric> batch, ICollection<MetricSnapshot> exportedItems)
+        {
+            if (exportedItems == null)
```

Comment:
given the Guards to the addinmemoryexporter methods, this should never be null

Reply:
In the `AddInMemoryExporter` method, we provide a reference to the collection. This could be set to `null` later on externally. We don't want `NullReferenceException` in that case.

Reply:
I agree with @utpilla here. restored the if check